examples/training/classification/imagenet/basic_training.py

```diff
@@ -193,6 +192,11 @@ def augment(img, label):
     inputs = {"images": img, "labels": label}
     for layer in AUGMENT_LAYERS:
         inputs = layer(inputs)
+    if tf.random().uniform(()) > 0.5:
+        inputs = keras_cv.layers.CutMix()(inputs)
+    else:
+        inputs = keras_cv.layers.MixUp()(inputs)
```

Comment:
Don't we have any functionality to achieve this? https://github.com/keras-team/keras-cv/issues/408
cc @LukeWood

Reply:
Maybe! [RandomAugmentationPipeline](https://keras.io/api/keras_cv/layers/preprocessing/random_augmentation_pipeline/)

Reply:
Good catch! Trying a run with:

```
AUGMENT_LAYERS = [
    keras_cv.layers.RandomFlip(mode="horizontal"),
    keras_cv.layers.RandAugment(value_range=(0, 255), magnitude=0.3),
    keras_cv.layers.RandomAugmentationPipeline(layers=[keras_cv.layers.CutMix(), keras_cv.layers.MixUp()], augmentations_per_image=1, rate=1)
]
```

Reply:
```
AUGMENT_LAYERS = [
    keras_cv.layers.RandomFlip(mode="horizontal"),
    keras_cv.layers.RandAugment(value_range=(0, 255), magnitude=0.3),
    keras_cv.layers.RandomAugmentationPipeline(layers=[keras_cv.layers.CutMix(), keras_cv.layers.MixUp()], augmentations_per_image=1, rate=1)
]

@tf.function
def augment(img, label):
    inputs = {"images": img, "labels": label}
    for layer in AUGMENT_LAYERS:
        inputs = layer(inputs)
        
    return inputs["images"], inputs["labels"]


train_set = (
    train_set.map(crop_and_resize, num_parallel_calls=tf.data.AUTOTUNE)
    .batch(BATCH_SIZE)
    .map(augment, num_parallel_calls=tf.data.AUTOTUNE)
    .prefetch(tf.data.AUTOTUNE)
)
```

🤔 

```
ValueError: CutMix received a single image to `call`.  The layer relies on combining multiple examples, and as such will not behave as expected.  Please call the layer with 2 or more samples.
```

Reply:
But works perfectly fine for the code in the PR, which should be functionally equivalent

Reply:
Some augmentation require batch of inputs, such as cut-mix, mix-up, fourier-mix, mosaic-mix.

```python
keras_cv.layers.RandomAugmentationPipeline(
   layers=[
        keras_cv.layers.CutMix(), 
        keras_cv.layers.MixUp()], 
       augmentations_per_image=1, # < shuld be > 1?
       rate=1 # is it ok? 
)
```

Reply:
Yes - CutMix and MixUp need batches of images and labels to mix up, but I think that we are providing batches? 
The initial code deals with the same batched input and they correctly work. I'll look into the potential difference in input

Reply:
The augmentations per image are set to 1 with a rate of 1 to ensure that on each augmentation call, we apply exactly 1 aug out of the two with a 50/50 probability